lib/langchain/llm/openai.rb

```diff
@@ -66,20 +66,29 @@ def complete(prompt:, **params)
     # @param context [String] The context of the conversation
     # @param examples [Array] Examples of messages provide model with
     # @param options extra parameters passed to OpenAI::Client#chat
+    # @param stream [Boolean] Stream the response true/false; Block must be provided as well.
     # @return [String] The chat completion
     #
-    def chat(prompt: "", messages: [], context: "", examples: [], **options)
+    def chat(prompt: "", messages: [], context: "", examples: [], stream: false, **options)
       raise ArgumentError.new(":prompt or :messages argument is expected") if prompt.empty? && messages.empty?
 
       parameters = compose_parameters DEFAULTS[:chat_completion_model_name], options
       parameters[:messages] = compose_chat_messages(prompt: prompt, messages: messages, context: context, examples: examples)
       parameters[:max_tokens] = validate_max_tokens(parameters[:messages], parameters[:model])
 
+      if (streaming = stream && block_given?)
```

Comment:
I think we can avoid having `stream` option and just look at `block_given?` So we'll call this method either like:

```ruby
result = llm.chat(prompt: "hello")
```

or 
```ruby
llm.chat(prompt: "hello") do |chunk|
   # do stuff
end
```